functions/nub.go

```diff
@@ -0,0 +1,18 @@
+package functions
+
+func (s SliceType) Nub() (out SliceType) {
```

Comment:
Looks good, but 2 changes I'd make:
* It can be more performant (one loop, better set with map[T]struct{})
* The order of the elements changed now. With a small change we can keep the input order :)
 
```go
contains := make (map[ElementType]struct{})
for _,v := range s {
     if _, ok := contains[v]; !ok {
             contains[v] = struct{}{}
             out = append(out, v)
     }
}
return
```